internal/evmreader/evmreader.go

```diff
@@ -147,113 +164,58 @@ func (r *EvmReader) watchForNewBlocks(ctx context.Context, ready chan<- struct{}
 		case err := <-sub.Err():
 			return &SubscriptionError{Cause: err}
 		case <-headers:
+
 			// Every time a new block arrives
-			err = r.checkForNewInputs(ctx)
+
+			// Get All Applications
+			runningApps, err := r.repository.GetAllRunningApplications(ctx)
 			if err != nil {
-				slog.Error("Error checking for new inputs",
+				slog.Error("Error retrieving running applications for new inputs",
 					"error",
 					err,
 				)
+				continue
 			}
 
-		}
-	}
-}
-
-// Check if is there new Inputs for all running Applications
-func (r *EvmReader) checkForNewInputs(ctx context.Context) error {
-
-	slog.Debug("Checking for new inputs")
-
-	// Get All Applications
-	apps, err := r.repository.GetAllRunningApplications(ctx)
-	if err != nil {
-		return err
-	}
-
-	if len(apps) == 0 {
-		slog.Info("No running applications")
-		return nil
-	}
-
-	groupedApps := r.classifyApplicationsByLastProcessedInput(apps)
-
-	for lastProcessedBlock, apps := range groupedApps {
-
-		appAddresses := appToAddresses(apps)
-
-		// Safeguard: Only check blocks starting from the block where the InputBox
-		// contract was deployed as Inputs can be added to that same block
-		if lastProcessedBlock < r.inputBoxDeploymentBlock {
-			lastProcessedBlock = r.inputBoxDeploymentBlock - 1
-		}
-
-		mostRecentHeader, err := r.fetchMostRecentHeader(
-			ctx,
-			r.defaultBlock,
-		)
-		if err != nil {
-			slog.Error("Error fetching most recent block",
-				"default block", r.defaultBlock,
-				"error", err)
-			continue
-		}
-		mostRecentBlockNumber := mostRecentHeader.Number.Uint64()
-
-		if mostRecentBlockNumber > lastProcessedBlock {
+			// Build Contracts
+			var apps []application
+			for _, app := range runningApps {
+				applicationContract, consensusContract, err := r.getAppContracts(app)
+				if err != nil {
+					slog.Error("Error retrieving application contracts", "app", app, "error", err)
+					continue
+				}
+				apps = append(apps, application{Application: app,
+					applicationContract: applicationContract,
+					consensusContract:   consensusContract})
+			}
 
-			slog.Info("Checking inputs for applications",
-				"apps", appAddresses,
-				"last processed block", lastProcessedBlock,
-				"most recent block", mostRecentBlockNumber,
-			)
+			if len(apps) == 0 {
```

Comment:
Applications could be left out because of IConsensus inconsistencies, since the blockchain registered IConsensus could be different from the DB registered one.
We should process only the "consistent" ones